src/pl/michaeli/spring/dao/ReservationsDAO.java

```diff
@@ -0,0 +1,67 @@
+package pl.michaeli.spring.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import javax.sql.DataSource;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.PreparedStatementCreator;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Component;
+
+import pl.michaeli.model.Reservation;
+import pl.michaeli.model.ReservationMapper;
+
+
+@Component
+public class ReservationsDAO {
+	JdbcTemplate jdbcTemplate;
+
+	private final String SQL_GET_RESERVATION_BY_ID = "SELECT * FROM RESERVATIONS WHERE ID = ?;";
+	private final String SQL_UPDATE_RESERVATION_BY_VOUCHER = "UPDATE RESERVATIONS SET VOUCHER_ID=?,TOTAL_AMOUNT=TOTAL_AMOUNT/2 WHERE ID=?";
```

Comment:
Dlaczego dzielisz na 2? I dlaczego w zapytaniu, a nie w kodzie?

Reply:
Voucher miał dawać -50%
A nie robiłam tego w kodzie, bo stwierdziłam, że może tak będzie wydajniej?
Bo z poziomu kodu muszę najpierw pobrać bieżącą kwotę, podzielić ją i dopiero update. A tak to tylko update leci